### ISSUE TITLE
How to read the scenario protos data in motion dataset?

### ISSUE THREAD

**Issue by j-sun-10** [waymo-research/waymo-open-dataset#358]
Can someone give me some ideas about how to read the data which is stored as **scenario protocol buffers**? I guess the tutorial is only valid for tf.example format. Thanks in advance.

**Comment by iyuner** [comment]
Hi! Maybe [this](https://github.com/waymo-research/waymo-open-dataset/issues/343) can help you

**Comment by j-sun-10** [comment]
Thanks a lot, it really works. I have a small question about compiling the scenario.proto file. For the definition of the message Scenario, the line `reserved 9` cannot be compiled well. The error is 
- scenario.proto:95:3: Expected "required", "optional", or "repeated".  
- scenario.proto:95:12: Expected field name.
I commented this line out to solve this problem. But I would ask if there is a better solution.

**Comment by scott-ettinger** [comment]
Reserved fields are standard in protocol buffers.  See : https://developers.google.com/protocol-buffers/docs/proto#reserved.  The only thing I can think of is if maybe you are using a very old protocol buffers package?

**Comment by 18627242758** [comment]
Is there a way to visualize scenario protos data? I think currently there are tutorials to visualize only tf.Example protos data.

**Comment by ares89** [comment]
> Is there a way to visualize scenario protos data? I think currently there are tutorials to visualize only tf.Example protos data.

I can read the scenario protos data, but I cannot find any information about converting from scenario protos data to tfexample format,

**Comment by ares89** [comment]
> We have not released the code to convert between the scenario and tf.Example formats. Can you tell me what you need for your application?

I want tf.Example formats data and have downloaded the scenarion formats files. Due to the unstable of network, downloading files is not easy. So I want to do converting.

**Comment by scott-ettinger** [comment]
Unfortunately, we are not releasing the conversion code at this time. We may provide it in future releases.  We provide the specification for our tf.Example conversion here : https://waymo.com/open/data/motion/tfexample.  Unfortunately you would have to write your own code to do the conversion.

**Comment by ares89** [comment]
> Unfortunately, we are not releasing the conversion code at this time. We may provide it in future releases. We provide the specification for our tf.Example conversion here : https://waymo.com/open/data/motion/tfexample. Unfortunately you would have to write your own code to do the conversion.

Got it. Thank you all the same.

**Comment by ZiffW** [comment]
> Unfortunately, we are not releasing the conversion code at this time. We may provide it in future releases. We provide the specification for our tf.Example conversion here : https://waymo.com/open/data/motion/tfexample. Unfortunately you would have to write your own code to do the conversion.

Hi! Just curious is there any tutorial code that we can visualize the scenario motion data? Thanks!

**Comment by EnnaSachdeva** [comment]
Hi,
How to identify which scenario in tf_example corresponds to which scenario in scenario protos? I noticed that the 'scenario/id' of tf_example doesn't match the 'scenario_id' of Scenario protos data format. Also, the visualization of tf_example and scenario protos show that scenarios are different. For instance- below is the visualizations of the very first scenario of training.tfrecord-00000-of-01000 (scenario proto data)
![temp_2](https://user-images.githubusercontent.com/15670305/186559440-c0dd2b45-876e-4e58-87d5-2da2514b57fd.PNG)
![temp_1](https://user-images.githubusercontent.com/15670305/186559442-14d2e025-08b5-4d91-9400-9a7bff484d43.PNG)
 and training_tfexample.tfrecord-00000-of-01000 (tf_example data).

**Comment by scott-ettinger** [comment]
The only way to match tf.Example data to Scenario protos is to find the records with matching IDs.  The two datasets do not store examples in the same order.  Could you explain more about your application and why you need to match them?

**Comment by EnnaSachdeva** [comment]
I want to extract specific scenarios from the datasets (for instance- car following). This extraction requires lane information, which is not present in the tf example data, that's why I am required to use scenario protos. Once I extract desired agents' IDs from scenarios protos, I plan to use those IDs in tf example dataset format to train my model. That's why I need to match the scenario in tf_example and scenario protos.

**Comment by scott-ettinger** [comment]
I see.  Currently this is a challenge.  We plan to open source the conversion code from Scenario protos to tf.Examples in the next release. This will make this task much easier as you can modify the code to add the additional roadgraph data to the tf.Examples.

**Comment by EnnaSachdeva** [comment]
Any timeline on when you plan the next release?

**Comment by scott-ettinger** [comment]
Yes, by matching I meant finding a tf.Example with a scenario/id field that matches the scenario_id field in the Scenario proto.  
As far as timeline for the next release, I cannot give any specific time.  I can say we are currently working on it.

**Comment by zhengzl18** [comment]
> I see. Currently this is a challenge. We plan to open source the conversion code from Scenario protos to tf.Examples in the next release. This will make this task much easier as you can modify the code to add the additional roadgraph data to the tf.Examples.

Any update on this? I'm currently using v1.2.1 but matching scenario and tf_example data still seems a challenge.
